### ISSUE TITLE
README Docs Issue

### ISSUE THREAD

**Issue by alanisaac** [bobthemighty/punq#47]
The README has this example:

> If our application's dependencies have their own dependencies, Punq will inject those, too:
```py
class Greeter:
   def greet(self):
      pass


class ConsoleGreeter:
   def __init__(self, config_reader: ConfigReader):
      self.config = config_reader.get_config()

   def greet(self):
      print(self.config['greeting'])


container.register(Greeter)
container.resolve(Greeter).greet()
```

The way the example is laid out, it doesn't seem like it's doing what the documentation above it would indicate.  Is the second to last line supposed to be:

```py
container.register(Greeter, ConsoleGreeter)
```

Otherwise `ConsoleGreeter` won't be resolved, is that correct?  (Happy to make a PR if this is the case!)

**Comment by jlecount** [comment]
Was just coming here to say the same --  I also presumed you wanted ConsoleGreeter to extend Greeter.  Since there's not a PR yet on this, will send one your way.

**Comment by jlecount** [comment]
Added PR here[here](https://github.com/bobthemighty/punq/pull/50) .

**Comment by bobthemighty** [comment]
Closed by #50
